### PR TITLE
Add tests and error messages

### PR DESCRIPTION
**Pull request by eeue56**
## What

- Adds new tests for loadDir and postache

### Auth and database url
- Has more extensive edge-case handling around auth and database URL errors
- If no auth is given, default to empty
- If username is given but no password, default to empty
- If an invalid URL is given, state why, and omit the password

ti 04 juli11 35 27 02002017

## Load dir
- `loadDir` previously was not filtering any results, now it does as intended.
- Makes `loadDir` only match the given extension at the end of the file instead of any part of the file name

## Why

- We ran into a couple of issues when using this library with invalid auth, and the error given was `conStr.auth.split(':') is not a function`

### CODE REVIEW

**Comment by eeue56** [comment]
How do you feel about using something like prettier? It does a pretty good job! We can even set up some options: https://github.com/prettier/prettier-eslint

**Comment by eeue56** [comment]
Added eslint and cleaned things it pointed out up :)

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module.exports = {
+    "extends": "airbnb-base",
+    "plugins": [
+        "import"
+    ],
+    "rules": {
+      "quotes": 0,
```
Comment (ahultgren) [inline .eslintrc.js]: How about `"quotes": [1, "double"],`?

**Comment by ahultgren** [comment]
About prettier i'm fine with it as long as it's able to mostly adhere to the same config as we use for content and importer etc. It seems with prettier-eslint it would. I wonder however how does it work together with nodemon, if it rewrites the files? Or i guess it only actually rewrites if the output has changed?

File: .eslintrc.js (hunk 1 above)
Comment (eeue56) [inline .eslintrc.js]: Force pushed with that config :+1:

**Comment by eeue56** [comment]
@ahultgren I gave prettier a try in this commit: https://github.com/omninews/node-postache/commit/61d79dfc046a89941269e9a9160121fdf96f6eee

The main difference looks like arrow-parens on single-arg arrow functions. Seems a bit weird to me but overall the benefits of an automatic formatter that works can't be over-stated :smile: 

Regarding how it'd work with nodemon, I think it would be fine -- it only touches files if they have changed, otherwise it only does a simple parse (no writes)

**Comment by ahultgren** [comment]
I was taught to use arg-parens when the function has a {}-body, but i don't really care either way. If you think it's better let's use it! The implementation doesn't matter much as long as it's run through `npm test`.

**Comment by eeue56** [comment]
I think it's worth trying out :) I'm not the best at formatting JS, and it can get really tricky when there's not a tool to do it all for you! We can maybe add it to this repo, and I'll take a look at how big the diff will be for others?

**Comment by eeue56** [comment]
Hm, I merged try-prettier into this branch and Github dismissed the review. Wasn't me!
